Latest version turns on pybindings and xnnpack everywhere, and coreml/mps on macos.

Linux build failing at https://github.com/pytorch/executorch/actions/runs/8676736127/job/23791536165?pr=2969#step:14:142
```
/__w/executorch/executorch/pytorch/executorch/kernels/portable/cpu/op_isinf.cpp: In function ‘torch::executor::Tensor& torch::executor::native::isinf_out(torch::executor::RuntimeContext&, const Tensor&, torch::executor::Tensor&)’:
/__w/executorch/executorch/pytorch/executorch/kernels/portable/cpu/op_isinf.cpp:18:71: error: cannot resolve overloaded function ‘isinf’ based on conversion to type ‘torch::executor::FunctionRef<bool(double)>’
   18 |   return internal::unary_ufunc_realhb_to_bool(std::isinf, ctx, in, out);
      |                                                                       ^
gmake[3]: *** [kernels/portable/CMakeFiles/portable_kernels.dir/cpu/op_isinf.cpp.o] Error 1
gmake[3]: *** Waiting for unfinished jobs....
/__w/executorch/executorch/pytorch/executorch/kernels/portable/cpu/op_isnan.cpp: In function ‘torch::executor::Tensor& torch::executor::native::isnan_out(torch::executor::RuntimeContext&, const Tensor&, torch::executor::Tensor&)’:
/__w/executorch/executorch/pytorch/executorch/kernels/portable/cpu/op_isnan.cpp:18:71: error: cannot resolve overloaded function ‘isnan’ based on conversion to type ‘torch::executor::FunctionRef<bool(double)>’
   18 |   return internal::unary_ufunc_realhb_to_bool(std::isnan, ctx, in, out);
      |                                                                       ^
gmake[3]: *** [kernels/portable/CMakeFiles/portable_kernels.dir/cpu/op_isnan.cpp.o] Error 1
gmake[2]: *** [kernels/portable/CMakeFiles/portable_kernels.dir/all] Error 2
gmake[1]: *** [CMakeFiles/portable_lib.dir/rule] Error 2
gmake: *** [portable_lib] Error 2
```

macos build failing in coreml https://github.com/pytorch/executorch/actions/runs/8676736126/job/23791536162?pr=2969#step:12:249
```
/Users/ec2-user/runner/_work/executorch/executorch/pytorch/executorch/backends/apple/coreml/runtime/inmemoryfs/inmemory_filesystem.cpp:139:75: error: 'path' is unavailable: introduced in macOS 10.15
InMemoryFileSystem::Attributes get_file_attributes(const std::filesystem::path& path) {
                                                                          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/filesystem:909:24: note: 'path' has been explicitly marked unavailable here
class _LIBCPP_TYPE_VIS path {
                       ^
/Users/ec2-user/runner/_work/executorch/executorch/pytorch/executorch/backends/apple/coreml/runtime/inmemoryfs/inmemory_filesystem.cpp:141:53: error: 'last_write_time' is unavailable: introduced in macOS 10.15
```
And similar for `directory_iterator`. So maybe we can't build coreml for this prebuilt on our CI mac system.

--- a/.github/workflows/build-wheels-linux.yml
+++ b/.github/workflows/build-wheels-linux.yml
@@ -54,6 +54,7 @@ jobs:
       # "recursive" default to do less work, and to give the buck daemon fewer
       # files to look at.
       submodules: true
+      env-var-script: build/packaging/env_var_script_linux.sh
       pre-script: ${{ matrix.pre-script }}
       post-script: ${{ matrix.post-script }}
       package-name: ${{ matrix.package-name }}

--- a/.github/workflows/build-wheels-m1.yml
+++ b/.github/workflows/build-wheels-m1.yml
@@ -54,6 +54,7 @@ jobs:
       # "recursive" default to do less work, and to give the buck daemon fewer
       # files to look at.
       submodules: true
+      env-var-script: build/packaging/env_var_script_m1.sh
       pre-script: ${{ matrix.pre-script }}
       post-script: ${{ matrix.post-script }}
       package-name: ${{ matrix.package-name }}

--- a/build/packaging/env_var_script_linux.sh
+++ b/build/packaging/env_var_script_linux.sh
@@ -1,0 +1,20 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# This file is sourced into the environment before building a pip wheel. It
+# should typically only contain shell variable assignments. Be sure to export
+# any variables so that subprocesses will see them.
+
+# Enable pybindings so that users can execute ExecuTorch programs from python.
+export EXECUTORCH_BUILD_PYBIND=1
+
+# Ensure that CMAKE_ARGS is defined before referencing it. Defaults to empty
+# if not defined.
+export CMAKE_ARGS="${CMAKE_ARGS:-}"
+
+# Link the XNNPACK backend into the pybindings runtime so that users can execute
+# ExecuTorch programs that delegate to it.
+CMAKE_ARGS="${CMAKE_ARGS} -DEXECUTORCH_BUILD_XNNPACK=ON"

--- a/build/packaging/env_var_script_m1.sh
+++ b/build/packaging/env_var_script_m1.sh
@@ -1,0 +1,28 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# This file is sourced into the environment before building a pip wheel. It
+# should typically only contain shell variable assignments. Be sure to export
+# any variables so that subprocesses will see them.
+
+# Enable pybindings so that users can execute ExecuTorch programs from python.
+export EXECUTORCH_BUILD_PYBIND=1
+
+# Ensure that CMAKE_ARGS is defined before referencing it. Defaults to empty
+# if not defined.
+export CMAKE_ARGS="${CMAKE_ARGS:-}"
+
+# Link the XNNPACK backend into the pybindings runtime so that users can execute
+# ExecuTorch programs that delegate to it.
+CMAKE_ARGS="${CMAKE_ARGS} -DEXECUTORCH_BUILD_XNNPACK=ON"
+
+# When building for macOS, link additional backends into the pybindings runtime.
+
+# TODO(dbort): Core ML requires features only available in macOS 10.15, but the
+# build machine uses an older version.
+# CMAKE_ARGS="${CMAKE_ARGS} -DEXECUTORCH_BUILD_COREML=ON"
+
+CMAKE_ARGS="${CMAKE_ARGS} -DEXECUTORCH_BUILD_MPS=ON"

--- a/kernels/portable/cpu/op_isinf.cpp
+++ b/kernels/portable/cpu/op_isinf.cpp
@@ -14,8 +14,18 @@ namespace torch {
 namespace executor {
 namespace native {
 
+namespace {
+// Passing std::isinf directly to unary_ufunc_realhb_to_bool can cause "error:
+// cannot resolve overloaded function ‘isinf’ based on conversion to type
+// ‘torch::executor::FunctionRef<bool(double)>’" in some compilation
+// environments.
+bool isinf_wrapper(double num) {
+  return std::isinf(num);
+}
+} // namespace
+
 Tensor& isinf_out(RuntimeContext& ctx, const Tensor& in, Tensor& out) {
-  return internal::unary_ufunc_realhb_to_bool(std::isinf, ctx, in, out);
+  return internal::unary_ufunc_realhb_to_bool(isinf_wrapper, ctx, in, out);
 }
 
 } // namespace native

--- a/kernels/portable/cpu/op_isnan.cpp
+++ b/kernels/portable/cpu/op_isnan.cpp
@@ -14,8 +14,18 @@ namespace torch {
 namespace executor {
 namespace native {
 
+namespace {
+// Passing std::isnan directly to unary_ufunc_realhb_to_bool can cause "error:
+// cannot resolve overloaded function ‘isnan’ based on conversion to type
+// ‘torch::executor::FunctionRef<bool(double)>’" in some compilation
+// environments.
+bool isnan_wrapper(double num) {
+  return std::isnan(num);
+}
+} // namespace
+
 Tensor& isnan_out(RuntimeContext& ctx, const Tensor& in, Tensor& out) {
-  return internal::unary_ufunc_realhb_to_bool(std::isnan, ctx, in, out);
+  return internal::unary_ufunc_realhb_to_bool(isnan_wrapper, ctx, in, out);
 }
 
 } // namespace native

--- a/setup.py
+++ b/setup.py
@@ -83,11 +83,6 @@ class ShouldBuild:
     def pybindings(cls) -> bool:
         return cls._is_env_enabled("EXECUTORCH_BUILD_PYBIND", default=False)
 
-    @classmethod
-    @property
-    def xnnpack(cls) -> bool:
-        return cls._is_env_enabled("EXECUTORCH_BUILD_XNNPACK", default=False)
-
 
 class _BaseExtension(Extension):
     """A base class that maps an abstract source to an abstract destination."""
@@ -372,13 +367,9 @@ class CustomBuild(build):
                 "-DEXECUTORCH_BUILD_PYBIND=ON",
             ]
             build_args += ["--target", "portable_lib"]
-            if ShouldBuild.xnnpack:
-                cmake_args += [
-                    "-DEXECUTORCH_BUILD_XNNPACK=ON",
-                ]
-                # No target needed; the cmake arg will link xnnpack
-                # into the portable_lib target.
-            # TODO(dbort): Add MPS/CoreML backends when building on macos.
+            # To link backends into the portable_lib target, callers should
+            # add entries like `-DEXECUTORCH_BUILD_XNNPACK=ON` to the CMAKE_ARGS
+            # environment variable.
 
         # Allow adding extra cmake args through the environment. Used by some
         # tests and demos to expand the set of targets included in the pip

